src/internal/metrics/index.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PACKAGE_VERSION, THEME } from '../environment';
+import { useVisualRefresh } from '../hooks/use-visual-mode';
+import { Metrics, MetricsTestHelper } from './metrics';
+import * as awsuiVersions from './init-awsui-versions';
+import * as componentMetrics from './use-component-metrics';
+
+export const metrics = new Metrics(PACKAGE_VERSION);
+export const metricsTestHelper = new MetricsTestHelper();
+
+export function initAwsUiVersions() {
+  awsuiVersions.initAwsUiVersions(PACKAGE_VERSION);
+}
+
+export function useComponentMetrics<T = any>(componentName: string) {
+  const theme = useVisualRefresh() ? 'vr' : THEME;
```

Comment:
This condition is a trait of all components in our system, including new the package. 

Actually, I would extract the original `useBaseComponent` hook into the shared utils, because it contains all features we need for all components. You may keep the hooks separately in the implementation, but the utility API should expose `useBaseComponent`

Reply:
Ok, the only difference between `useComponentMetrics` and `useBaseComponent` is the theme and package version injection. The package injection will still happen from outside the toolkit but the theme can be handled of course. The downside is that we'd have to move or copy at least the `useVisualRefresh` hook to the toolkit to support that.